src/js/components/pipelines/HoverField.jsx

```diff
@@ -110,15 +117,41 @@ var HoverField = React.createClass({
     this.setState({fieldDef: fieldDef});
   },
 
+  sortValues: function(evt) {
+  // will need to clean up this code
+  var valuesInc = this.state.valuesInc,
+      props = this.props,
+      field = props.def,
+      id = props.dsId,
+      incOrDec = null;
+
+  if (valuesInc == null) {
+    // first click default: increasing
+    valuesInc = true;
+  } else {
+    valuesInc = !valuesInc;
+  }
+  this.setState({ valuesInc : valuesInc });
+
+  incOrDec = valuesInc ? 'inc' : 'dec';
+  this.props.sortDataset(id, field.name, incOrDec);
+},
+
   render: function() {
     var state = this.state,
         field = state.fieldDef,
-        style = {top: state.offsetTop, display: field ? 'block' : 'none'};
+        style = {top: state.offsetTop, display: field ? 'block' : 'none'},
+        incOrDecIcon = this.state.valuesInc ? (
+          <Icon onClick={this.sortValues}
+            glyph={assets['increasingSort']} width="10" height="10" /> ) : (
+          <Icon onClick={this.sortValues}
+            glyph={assets['decreasingSort']} width="10" height="10" /> );
 
     field = field ? (
       <div>
         <FieldType field={field} />
         {field.name}
+        {incOrDecIcon}
```

Comment:
Can you extract all the changes you've made here into a separate `SortField.jsx` component? The hope would be to a fairly standalone/self-contained component, much like we did with `FieldType.jsx`. Unless I'm mistaken, the `SortField` component should not require any local state (i.e., `this.state` within the react component itself). Instead, we should be able to drive it entirely based on what is stored in the `dataset._sort` in the store.